prelude/slang-cuda-prelude.h

```diff
@@ -5,6 +5,9 @@
 // are passed down. 
 
 #ifdef SLANG_CUDA_ENABLE_HALF
+// We don't want half2 operators, because it will implement comparison operators that return a bool(!). We want to generate
+// those functions. Doing so means that we will have to define all the other half2 operators.
+#   define __CUDA_NO_HALF2_OPERATORS__
```

Comment:
It is possible that we can/should emit all the vector/matrix comparison operators for C++ and CUDA as function calls rather than directly as operators.

Reply:
We could but would end up being more complicated - as we'd need to make the 'generators' able to know about CUDA specifics. 

What would be *nice* would be more flexible/powerful ways to build types out of 'builtin' types targets have. Ideally I could implement half and half2 and then everything else could be inferred (with some other info perhaps) from that.